### PR TITLE
Move logging to @ConfigMapping

### LINKED ISSUE #42114: Core module is using a mix of traditional `@ConfigRoot` and `@ConfigMapping`
@radcortez if you want to help, it would be nice if you could make sure the `quarkus-core` modules (runtime and deployment) uniformly use `@ConfigMapping` (or `@ConfigRoot`, your choice but I suspect you will choose the former).

This won't be allowed anymore with the new annotation processor.

See `PackageConfig` for `@ConfigMapping` and `ShutdownBuildTimeConfig` for traditional `@ConfigRoot`.

### PR DESCRIPTION
**Pull request by radcortez**
- Partially Fixes #42114

It would help to come up with fixes for:
- https://github.com/quarkusio/quarkus/issues/35693
- https://github.com/quarkusio/quarkus/issues/43180

Noticeable changes:
- The color hack in https://github.com/quarkusio/quarkus/pull/42157/files#diff-336136f742a05498477b4a3dd2b63bf198237f763c1a336ea654f3bc499e2c05L70-L80 is no longer required, since `@ConfigMapping` allows you to map the same configuration multiple times (the `@ConfigRoot` class implementation only allowed one matching property). So, `quarkus.console.color` is now available explicitly in both build time and runtime.
- The `LogRuntimeConfig` is also used during build time, but the `@ConfigRoot` limitation required the `ConfigInstantiator` to create it. Since `LogBuildTimeConfig`, `LogRuntimeConfig`, and `ConsoleRuntimeConfig` are required to set the build time and the runtime log, they are always registered by default, meaning they can be retrieved directly from `SmallRyeConfig` without any additional hacks
- Rewrote the pieces that set up the log categories because the code was mutating the configuration directly: https://github.com/quarkusio/quarkus/pull/42157/files#diff-4c8e61042dd080918dbcd9915ff27fda148515edc6359b7d580417deadba937cL318-L349. This is not possible anymore with `@ConfigMapping`.
- Same for log filters: https://github.com/quarkusio/quarkus/pull/42157/files#diff-3027189c5073a902729f033232338719972cd18a3c64018f24b99dd78e1fea6aL296-L304

### CODE REVIEW

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks! Let's not merge this too quickly.

I wonder if 3.14 is the right version to merge this, or it would be better to wait for 3.16 to get branched.

For now, I implemented a workaround to support mixed extensions.

**Comment by geoand** [comment]
There is also a conflict in this

File: integration-tests/vertx/src/main/resources/application.properties [hunk 1]
```diff
@@ -1,3 +1,3 @@
 vertx.event-loops.size=2
 quarkus.log.file.enable=true
-quarkus.log.file=quarkus.log
\ No newline at end of file
+quarkus.log.file.path=quarkus.log
```
Comment (gsmet) [inline integration-tests/vertx/src/main/resources/application.properties:3]: This was a mistake in the config file or something changed?

**Review by gsmet — APPROVED** [review]
Let's try to get this in for 3.16.

I rebased to get a fresh CI run.

@radcortez can you just have a look at my question?

**Comment by gsmet** [comment]
I fixed an import conflict. Let's wait for CI and merge!

**Comment by radcortez** [comment]
Thanks!

**Pull request merged**